pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -71,6 +75,19 @@ type Registry struct {
 	Prefix string `toml:"prefix"`
 }
 
+// EndpointsFromReference returns a slice of Endpoints based on the passed
+// reference.
+func (r *Registry) EndpointsFromReference(ref reference.Named) []Endpoint {
```

Comment:
Please combine this with `RewriteReference` to have something like
```go
type RewrittenEndpoint struct {
    Endpoint *Endpoint // Or not a pointer, not sure
    Reference reference.Named
}

func (r *Registry) EndpointsForPulling(ref reference.Named) ([]RewrittenEndpoint, error)
```
, ideally with a better name than `RewrittenEndpoint` (`PullSource`? Not much better.)
(And then remove the public `RewriteReference`.)

Reply:
Nothing that can't be solved with another level of indirection :-)

Reply:
Is the “level of indirection” comment targeted at the naming? I’m not sure I understand.

Reply:
I should not make jokes in comments, apologies.

Reply:
`PullSource` sounds good. I could not come up with anything better.